specification/attestation.adoc

```diff
@@ -1,309 +1,880 @@
+attestation.
+2. The TSM TCB elements are the confidential compute specific but TVM agnostic
+elements for the overall TCB.
+3. The TVM TCB is composed of the TVM specific elements.
 
-The TSM must then provide an implementation of a TEE-Guest ABI (TG-ABI) 
-operation (sbi_tee_guest_get_evidence) to enable a TVM to generate 
-attestation evidence that a relying party can verify using the certificate 
-chain. 
+The TCB elements for each of them is summarized in the following table:
 
-The TCB extension and evidence collection for a TVM attestation is shown 
-below:
+.AP-TEE Workload TCB Elements
+[cols="^1, ^5v, ^4v", options="header"]
+|===
+|TCB |Hardware Elements |Software Elements
+
+.4+.^|Platform
+    <| HW RoT for boot, measurement and storage
+ .4+<| All M-mode firmwares, including the TSM-driver
+    <| All CPU hardware logic, including MMU and caches
+    <| All SoC subsystems, including memory confidentiality, integrity and replay-protection for volatile memory
+    <| IOMMU and translation agents
+
+     | TSM
+     | N/A
+     | TEE Security Manager (TSM) and its U-mode components
+
+     | TVM
+     | Directly assigned, TEE-IO compliant devices
+     | All TVM measured pages
+|===
 
-image:img_5.png[]
-Figure 5: Layered Attestation architecture for TVMs
+==== TVM Measurements
 
-It is expected that an implementation will provide implementation-specific 
-intrinsics to record measurements of the TSM into the firmware RoT for 
-measurement to support the layered RTMs and attestation of AP-TEE workloads. 
+At a high level, a TVM measurement is separated into a set of static
+measurements and a dynamic one.
 
-*Attestation Evidence*
+===== Static Measurements
 
-Suitable evidence formats may be used by the Attester to present the 
-evidence that the TVM is executing as a TEE. The evidence should attest to 
-the above layered trust chain. The TSM must allow for attestation operation 
-(certifying TVM measurements) to be executed in an interruptible manner. 
-Once such evidence format is specified in the TCG DICE Attestation 
-Architecture which describes evidence as X.509 Certificate with an 
-extension for *TCB Info Evidence <<R2>>. 
+TVM static measurements are generated from the AP-TEE workload TCB elements
+involved in the TVM construction. Any TCB element that directly or indirectly
+supports a TVM must be measured into the TVM static measurement registers. Once
+a TVM is finalized, i.e. after the `sbi_tee_host_finalize_tvm()` TH-ABI is
+called, the TVM static measurements must no longer be extended.
 
-The following key fields are present in that DiceTcbInfo (See OID in spec 
-<<R2>>). The fields are listed here with the usage described specific to the 
-AP-TEE reference architecture.
+Static measurements are stored in dedicated measurement registers and an AP-TEE
+implementation must provide at least 1 and at most 8 of them. The static
+measurement registers cover platform, TSM & TVM specific TCB element
+measurements.
```

Comment:
is this minimum 1 to max 8 for each TVM? then should qualify it as "TVM static measurements"

Reply:
Thanks, fixed.